lib/postgrex/notifications.ex

```diff
@@ -69,13 +69,20 @@ defmodule Postgrex.Notifications do
 
   require Logger
 
-  defstruct [:from, :ref, connected: false, listeners: %{}, listener_channels: %{}]
+  defstruct [
+    :from,
+    :ref,
+    auto_reconnect: false,
```

Comment:
Tracking the `auto_reconnect` status now so we only reply with `:eventually` on query error when auto_reconnect is true.